winrm/errors.go

```diff
@@ -0,0 +1,33 @@
+package winrm
+
+import (
+	"fmt"
+	"strings"
+)
+
+type ServerError struct {
```

Comment:
I think we don't need multiple return struct types and I think it's wrong that we added in the Client struct the err attribute.What I'm seeing  is ,  these types in order to be compatible with the error type in go it needs to implement the Error interface/contract.
```go
type error interface {
    Error() string
}
```
As I'm seeing this package needs desperately a **log** system or a good error descriptive general **error type**. Also it needs to separate the business logic and the error logic clearly and inserting that **err** field in the **client** type  is not in **my opinion** a good way.